plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployPropertyPage.java

```diff
@@ -105,69 +110,30 @@ public void run() {
 
   @Override
   public boolean performOk() {
-    // Content can be null if Properties Page is opened and App Engine facets are uninstalled
     if (isValid()) {
-      if (content == null) {
-        return true;
-      } else {
-        return content.savePreferences();
-      }
+      return getActivePanel().savePreferences();
     }
     return false;
   }
 
   @Override
   protected void performDefaults() {
-    content.resetToDefaults();
+    // should we reset all panels?
+    getActivePanel().resetToDefaults();
     super.performDefaults();
   }
 
   @Override
   public void dispose() {
-    if (content != null) {
-      // Content can be null if Properties Page is opened and App Engine facets are uninstalled
-      content.dispose();
-    }
+    blankPreferencesPanel.dispose();
+    standardPreferencesPanel.dispose();
+    flexPreferencesPanel.dispose();
```

Comment:
Currently the panel for Standard disposes the DBC in it's dispose(), the other two do not dispose it. So I need to wire those up as well.
What's better: override dispose() and call DBC.dispose() or add a listener? The listener would come handy if the DBCs would be injected into the panels, right?